### PR TITLE
JNG-4297 Creating Judo test methodology

### PR DESCRIPTION
**Pull request by fmagnucz**
<table>
<td>
  <a href="https://blackbelt.atlassian.net/browse/JNG-4297" title="JNG-4297" target="_blank">StoryJNG-4297</a>  Creating Judo test methodology
  </td></table>
  <br />

### CODE REVIEW

File: judo-runtime-core-jsl-itest/pom.xml [hunk 1]
```diff
@@ -425,6 +425,20 @@
                     -->
                 </executions>
             </plugin>
+
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>3.10.1</version>
```
Comment (robertcsakany) [inline judo-runtime-core-jsl-itest/pom.xml]: Remove version - already defined in parent

File: judo-runtime-core-jsl-itest/pom.xml [hunk 2]
```diff
@@ -425,6 +425,20 @@
...
+                <configuration>
+                    <source>11</source>
+                    <target>11</target>
```
Comment (robertcsakany) [inline judo-runtime-core-jsl-itest/pom.xml]: Remove source and target - already defined in parent

File: judo-runtime-core-jsl-itest/pom.xml [hunk 3]
```diff
@@ -488,5 +502,20 @@
...
+            <scope>compile</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>hu.blackbelt.judo</groupId>
+            <artifactId>judo-requirement-report-processor</artifactId>
+            <version>1.0.0.20230105_022024_241fcd45_feature_JNG_4297_Creating_Judo_test_methodology</version>
+            <optional>true</optional>
+            <scope>compile</scope>
+        </dependency>
```
Comment (robertcsakany) [inline judo-runtime-core-jsl-itest/pom.xml]: Do not use direct version reference in dependency for our nodule.

1. Please create a property in parent pom.xml in block 1.0.0.20230105_022024_241fcd45_feature_JNG_4297_Creating_Judo_test_methodolog
2. Put this dependency to parent's pom section with version using the property defined earlier ${judo-requirement-report-version}
3. In this place use the dependency without version and scope.
4. When a new property is defined in a parent's pom referencing our module, in judo-ng's (and judo-community too in this case) project-meta.yaml have to add the module as dependency. In our case in the judo-runtime-core JSL entry will have a new dependency which is judo-requirement-report.

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/FunctionsTest.java [hunk 4]
```diff
@@ -1,28 +1,6 @@
...
- * This Source Code may also be made available under the following Secondary
- * Licenses when the conditions for such availability set forth in the Eclipse
- * Public License, v. 2.0 are satisfied: GNU General Public License, version 2
- * with the GNU Classpath Exception which is
- * available at https://www.gnu.org/software/classpath/license.html.
- * 
- * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
- * #L%
- */
-
```
Comment (robertcsakany) [inline judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/FunctionsTest.java]: Why the licence block have beem removed? Please add back!

**Review by robertcsakany — CHANGES_REQUESTED** [review]

**Review by gaborflorian — APPROVED** [review]

**Review by bencegelei — APPROVED** [review]

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/FunctionsTest.java (hunk 4 above)
Comment (fmagnucz) [reply judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/FunctionsTest.java]: The merge process was the wrong.

**Review by gaborflorian — APPROVED** [review]

**Review by bencegelei — APPROVED** [review]

**Review by noherczeg — APPROVED** [review]

**Review by robertcsakany — APPROVED** [review]

**Pull request merged**